coverage_comment/coverage.py

```diff
@@ -88,20 +88,14 @@ class DiffCoverage:
 def compute_coverage(
     num_covered: int,
     num_total: int,
-    use_branch_coverage: bool,
-    num_branches_covered: int | None,
-    num_branches_total: int | None,
+    num_branches_covered: int = 0,
+    num_branches_total: int = 0,
 ) -> decimal.Decimal:
     """Compute the coverage percentage, with or without branch coverage."""
-    if use_branch_coverage:
-        num_branches_covered = cast(int, num_branches_covered)
-        num_branches_total = cast(int, num_branches_total)
-        numerator = decimal.Decimal(num_covered + num_branches_covered)
-        denominator = decimal.Decimal(num_total + num_branches_total)
-    else:
-        numerator = decimal.Decimal(num_covered)
-        denominator = decimal.Decimal(num_total)
-
+    num_branches_covered = cast(int, num_branches_covered)
+    num_branches_total = cast(int, num_branches_total)
```

Comment:
Cast is unneeded now :) (and pre-commit will tell you but the import of cast at the top of the file is likely unneeded too, then)

Honnestly, having to use `cast` is a code smell. The simple fact that you're able to remove it is proof enough that it's the right decision :)

But I'll even go further: if we want to make sure that branch coverage is supported everywhere, maybe `num_branches_covered` and `num_branches_total` should not be optional.

Reply:
> But I'll even go further: if we want to make sure that branch coverage is supported everywhere, maybe `num_branches_covered` and `num_branches_total` should not be optional.

Well, we _could_, but since I'm not using it in the diff computations, defaulting it here means I can call `compute_coverage()` **there** without having to supply explicit `0`s, which I kind of liked. So it goes both ways.

I also liked that I didn't have to change the `test_compute_coverage()` code at all, just add a separate `test_compute_coverage_with_branches()` that supplies branch values.

But I'm open to either, it's not a hill I'm looking to die on.

Reply:
Nah, I'm ok with an opinion sufficiently backed by good arguments. It's not like it's a decision we can't revisit later either so let's go your way.